diff --git a/c10/cuda/CUDACachingAllocator.cpp b/c10/cuda/CUDACachingAllocator.cpp
index 05941947ed..c8e0bafe0a 100644
--- a/c10/cuda/CUDACachingAllocator.cpp
+++ b/c10/cuda/CUDACachingAllocator.cpp
@@ -1141,7 +1141,7 @@ static std::string reportProcessMemoryInfo(int device) {
       DriverAPI::get()->nvmlDeviceGetHandleByPciBusId_v2_(
           pci_id, &nvml_device));
 
-  std::vector<nvmlProcessInfo_v1_t> procs(8);
+  std::vector<nvmlProcessInfo_t> procs(8);
   unsigned int size = procs.size();
   nvmlReturn_t r;
   while ((r = DriverAPI::get()->nvmlDeviceGetComputeRunningProcesses_(
