From f3a0f9aab6dce56eea590b946f60256014b61bf7 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 1 Sep 2024 17:35:40 -0400
Subject: [PATCH 01/18] Force usage of python 3 and error without numpy

---
 cmake/Dependencies.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index e78305e0a8e..15c62548601 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -861,9 +861,9 @@ if(BUILD_PYTHON)
   if(USE_NUMPY)
     list(APPEND PYTHON_COMPONENTS NumPy)
   endif()
-  find_package(Python COMPONENTS Interpreter OPTIONAL_COMPONENTS ${PYTHON_COMPONENTS})
+  find_package(Python REQUIRED COMPONENTS Interpreter ${PYTHON_COMPONENTS})
 else()
-  find_package(Python COMPONENTS Interpreter)
+  find_package(Python REQUIRED COMPONENTS Interpreter)
 endif()
 
 if(NOT Python_Interpreter_FOUND)
@@ -880,7 +880,7 @@ if(BUILD_PYTHON)
   if(Python_Development.Module_FOUND)
     if(USE_NUMPY)
       if(NOT Python_NumPy_FOUND)
-        message(WARNING "NumPy could not be found. Not building with NumPy. Suppress this warning with -DUSE_NUMPY=OFF")
+          message(FATAL_ERROR "NumPy could not be found. Not building with NumPy. Suppress this warning with -DUSE_NUMPY=OFF")
         caffe2_update_option(USE_NUMPY OFF)
       else()
         caffe2_update_option(USE_NUMPY ON)
