From 0c8d24175f9ec3e2cb05b755bbf19f953889369c Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Mon, 6 Nov 2023 22:00:51 -0500
Subject: [PATCH] Allow splayed layouts

---
 cmake/public/cuda.cmake | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/cmake/public/cuda.cmake b/cmake/public/cuda.cmake
index 32f3ba375b5..14318d7b399 100644
--- a/cmake/public/cuda.cmake
+++ b/cmake/public/cuda.cmake
@@ -60,13 +60,6 @@ find_package(CUDAToolkit REQUIRED)
 
 cmake_policy(POP)
 
-if(NOT CMAKE_CUDA_COMPILER_VERSION STREQUAL CUDAToolkit_VERSION OR
-    NOT CUDA_INCLUDE_DIRS STREQUAL CUDAToolkit_INCLUDE_DIR)
-  message(FATAL_ERROR "Found two conflicting CUDA installs:\n"
-                      "V${CMAKE_CUDA_COMPILER_VERSION} in '${CUDA_INCLUDE_DIRS}' and\n"
-                      "V${CUDAToolkit_VERSION} in '${CUDAToolkit_INCLUDE_DIR}'")
-endif()
-
 if(NOT TARGET CUDA::nvToolsExt)
   message(FATAL_ERROR "Failed to find nvToolsExt")
 endif()
-- 
2.42.0

