From 3fb6b3704a6359521e186bfd4c6644a56aa08d90 Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Thu, 17 Oct 2024 15:04:05 -0700
Subject: [PATCH 03/16] Update sympy version

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index a6a6db7b033..d73cec7dd86 100644
--- a/setup.py
+++ b/setup.py
@@ -1099,7 +1099,7 @@ def main():
         "filelock",
         "typing-extensions>=4.10.0",
         'setuptools ; python_version >= "3.12"',
-        'sympy==1.13.1 ; python_version >= "3.9"',
+        'sympy>=1.13.1,!=1.13.2 ; python_version >= "3.9"',
         "networkx",
         "jinja2",
         "fsspec",
