From be785be20dab23d5cee88e13adf40150ce9ead3c Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Sun, 3 Nov 2024 01:12:36 -0700
Subject: [PATCH 04/19] Fix duplicate linker script

---
 setup.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index d73cec7dd86..75fdfce7e35 100644
--- a/setup.py
+++ b/setup.py
@@ -1125,7 +1125,9 @@ def main():
             filein="cmake/prioritized_text.txt", fout="cmake/linker_script.ld"
         )
         linker_script_path = os.path.abspath("cmake/linker_script.ld")
-        os.environ["LDFLAGS"] = os.getenv("LDFLAGS", "") + f" -T{linker_script_path}"
+        ldflags = os.getenv("LDFLAGS", "")
+        if f"-T{linker_script_path}" not in ldflags:
+            os.environ["LDFLAGS"] = ldflags + f" -T{linker_script_path}"
         os.environ["CFLAGS"] = (
             os.getenv("CFLAGS", "") + " -ffunction-sections -fdata-sections"
         )
