From 822ea7904aa8ccb50239e7f42b98595cc537940f Mon Sep 17 00:00:00 2001
From: Bas Zalmstra <bas@prefix.dev>
Date: Thu, 16 May 2024 10:46:49 +0200
Subject: [PATCH 04/17] Fix FindOpenBLAS

---
 cmake/Modules/FindOpenBLAS.cmake | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/cmake/Modules/FindOpenBLAS.cmake b/cmake/Modules/FindOpenBLAS.cmake
index 21ae9e2521e..b4aeeef4012 100644
--- a/cmake/Modules/FindOpenBLAS.cmake
+++ b/cmake/Modules/FindOpenBLAS.cmake
@@ -36,22 +36,25 @@ IF(DEFINED ENV{OpenBLAS_LIB_NAME})
   SET(Open_BLAS_LIB_NAME $ENV{OpenBLAS_LIB_NAME})
 ENDIF()
 
-FIND_PATH(OpenBLAS_INCLUDE_DIR NAMES cblas.h PATHS ${Open_BLAS_INCLUDE_SEARCH_PATHS})
-FIND_LIBRARY(OpenBLAS_LIB NAMES ${Open_BLAS_LIB_NAME} PATHS ${Open_BLAS_LIB_SEARCH_PATHS})
-
 SET(OpenBLAS_FOUND ON)
 
 #    Check include files
+FIND_PATH(OpenBLAS_INCLUDE_DIR NAMES cblas.h PATHS ${Open_BLAS_INCLUDE_SEARCH_PATHS})
 IF(NOT OpenBLAS_INCLUDE_DIR)
     SET(OpenBLAS_FOUND OFF)
     MESSAGE(STATUS "Could not find OpenBLAS include. Turning OpenBLAS_FOUND off")
 ENDIF()
 
 #    Check libraries
-IF(NOT OpenBLAS_LIB)
+FOREACH(LIB lapack cblas blas)
+  FIND_LIBRARY(FOUND_LIB_${LIB} NAMES ${LIB} PATHS ${Open_BLAS_LIB_SEARCH_PATHS})
+  IF(NOT FOUND_LIB_${LIB})
     SET(OpenBLAS_FOUND OFF)
-    MESSAGE(STATUS "Could not find OpenBLAS lib. Turning OpenBLAS_FOUND off")
-ENDIF()
+    MESSAGE(STATUS "Could not find OpenBLAS lib ${LIB}. Turning OpenBLAS_FOUND off")
+  ELSE()
+    LIST(APPEND OpenBLAS_LIB ${FOUND_LIB_${LIB}})
+  ENDIF()
+ENDFOREACH()
 
 IF (OpenBLAS_FOUND)
   IF (NOT OpenBLAS_FIND_QUIETLY)
