From cc95a56d612815f171ba7e71d921f66abfcf2a81 Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Thu, 17 Oct 2024 15:04:05 -0700
Subject: [PATCH 04/12] Update sympy version

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 7174777ed4..65be34e39b 100644
--- a/setup.py
+++ b/setup.py
@@ -1158,7 +1158,7 @@ def main():
         "typing-extensions>=4.8.0",
         'setuptools ; python_version >= "3.12"',
         'sympy==1.12.1 ; python_version == "3.8"',
-        'sympy==1.13.1 ; python_version >= "3.9"',
+        'sympy>=1.13.1,!=1.13.2 ; python_version >= "3.9"',
         "networkx",
         "jinja2",
         "fsspec",
