diff --git a/setup.py b/setup.py
index 2b0cfa99d7..ae9f88ff34 100644
--- a/setup.py
+++ b/setup.py
@@ -1144,7 +1144,7 @@ def main():
         "typing-extensions>=4.8.0",
         'setuptools ; python_version >= "3.12"',
         'sympy==1.12.1 ; python_version == "3.8"',
-        'sympy==1.13.1 ; python_version >= "3.9"',
+        'sympy>=1.13.1,!=1.13.2 ; python_version >= "3.9"',
         "networkx",
         "jinja2",
         "fsspec",
