From 81de6b92a50ed43323f54e769f1b7c0661ebfb0e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Mon, 17 Mar 2025 15:03:30 +0100
Subject: [PATCH] Fix CUPTI lookup to include target directory

---
 cmake/Dependencies.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index c4661e39e..355a5c40f 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1605,9 +1605,10 @@ if(USE_KINETO)
         ${CUDA_SOURCE_DIR}/extras/CUPTI/lib64
         ${CUDA_SOURCE_DIR}/lib
         ${CUDA_SOURCE_DIR}/lib64
-        NO_DEFAULT_PATH)
+        )
 
     find_path(CUPTI_INCLUDE_DIR cupti.h PATHS
+        ${CUDAToolkit_TARGET_DIR}/include
         ${CUDA_SOURCE_DIR}/extras/CUPTI/include
         ${CUDA_INCLUDE_DIRS}
         ${CUDA_SOURCE_DIR}
