From 0737bb37ed1b4a58c915caba8ce6e9d692fee9e4 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 14 Jun 2025 07:34:48 +1100
Subject: [PATCH 09/18] skip test_norm_matrix_degenerate_shapes on numpy >=2.3

---
 test/test_linalg.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/test_linalg.py b/test/test_linalg.py
index 346a6c02044..6da0acdbc71 100644
--- a/test/test_linalg.py
+++ b/test/test_linalg.py
@@ -2118,6 +2118,7 @@ class TestLinalg(TestCase):
     @skipIf(np.lib.NumpyVersion(np.__version__) < '2.3.0', 'Numpy changed handling of degenerate inputs in 2.3.0')
     @skipCUDAIfNoMagma
     @skipCPUIfNoLapack
+    @unittest.skipIf(np.lib.NumpyVersion(np.__version__) >= "2.3.0", "Error behaviour changed")
     @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
     def test_norm_matrix_degenerate_shapes(self, device, dtype):
         def run_test_case(input, ord, dim, keepdim, should_error):
