From e0e5a5bf75e7033268ece4627d2f30556c31bc4d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Thu, 6 Mar 2025 13:57:25 +0100
Subject: [PATCH 13/16] Fix CUPTI lookup to include target directory

---
 cmake/Dependencies.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 1dee5889d34..11577a2c0dc 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1637,6 +1637,7 @@ if(USE_KINETO)
     endif()
 
     find_library(CUPTI_LIBRARY_PATH ${CUPTI_LIB_NAME} PATHS
+        ${CUDAToolkit_TARGET_DIR}/lib
         ${CUDA_SOURCE_DIR}
         ${CUDA_SOURCE_DIR}/extras/CUPTI/lib64
         ${CUDA_SOURCE_DIR}/lib
@@ -1644,6 +1645,7 @@ if(USE_KINETO)
         NO_DEFAULT_PATH)
 
     find_path(CUPTI_INCLUDE_DIR cupti.h PATHS
+        ${CUDAToolkit_TARGET_DIR}/include
         ${CUDA_SOURCE_DIR}/extras/CUPTI/include
         ${CUDA_INCLUDE_DIRS}
         ${CUDA_SOURCE_DIR}
-- 
2.45.2

