From b3371dcb971658239ce8aa8f6fa0a4d7bcea6735 Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Sat, 22 Mar 2025 22:50:49 -0700
Subject: [PATCH 14/17] Always use system nvtx

---
 cmake/public/cuda.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/public/cuda.cmake b/cmake/public/cuda.cmake
index e538fb5662b..b409c6c30e9 100644
--- a/cmake/public/cuda.cmake
+++ b/cmake/public/cuda.cmake
@@ -146,7 +146,7 @@ else()
 endif()
 
 # nvToolsExt
-if(USE_SYSTEM_NVTX)
+if(TRUE)
   find_path(nvtx3_dir NAMES nvtx3 PATHS ${CUDA_INCLUDE_DIRS})
 else()
   find_path(nvtx3_dir NAMES nvtx3 PATHS "${PROJECT_SOURCE_DIR}/third_party/NVTX/c/include" NO_DEFAULT_PATH)
