diff --git a/torch/testing/_internal/common_cuda.py b/torch/testing/_internal/common_cuda.py
index 01eeac86ae..2ded85e9a4 100644
--- a/torch/testing/_internal/common_cuda.py
+++ b/torch/testing/_internal/common_cuda.py
@@ -234,7 +234,7 @@ def _get_magma_version():
     return tuple(int(x) for x in version_str.split("."))
 
 def _get_torch_cuda_version():
-    if torch.version.cuda is None:
+    if torch.version.cuda is None or str(torch.version.cuda) == "None":
         return (0, 0)
     cuda_version = str(torch.version.cuda)
     return tuple(int(x) for x in cuda_version.split("."))
