From 2d7566334da792a27f7a01f6e8ac34975915156c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Mon, 3 Nov 2025 14:59:04 +0100
Subject: [PATCH 15/15] Fix building kineto against system fmt
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michał Górny <mgorny@gentoo.org>
---
 cmake/Dependencies.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 8b3a1b2629a..b99fbc5999f 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1597,6 +1597,8 @@ else()
     message(FATAL_ERROR "Cannot find system fmt library. Please install libfmt-dev or set USE_SYSTEM_FMT=OFF")
   endif()
   list(APPEND Caffe2_DEPENDENCY_LIBS fmt::fmt)
+  # for kineto
+  add_custom_target(fmt)
 endif()
 
 # ---[ Kineto
