From 250de80e0cb01749b089758c9cae92e46a6ca1f5 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 14 Jun 2025 07:34:48 +1100
Subject: [PATCH 16/16] skip test_norm_matrix_degenerate_shapes on numpy >=2.3

---
 test/test_linalg.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/test_linalg.py b/test/test_linalg.py
index 8ac5a198861..2371ebcbf0a 100644
--- a/test/test_linalg.py
+++ b/test/test_linalg.py
@@ -1866,6 +1866,7 @@ class TestLinalg(TestCase):
     # Test degenerate shape results match numpy for linalg.norm matrix norms
     @skipCUDAIfNoMagma
     @skipCPUIfNoLapack
+    @unittest.skipIf(np.lib.NumpyVersion(np.__version__) >= "2.3.0", "Error behaviour changed")
     @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
     def test_norm_matrix_degenerate_shapes(self, device, dtype):
         def run_test_case(input, ord, dim, keepdim, should_error):
