diff --git a/setup.py b/setup.py
index 9ec41cd37cd..5d1952b67b0 100644
--- a/setup.py
+++ b/setup.py
@@ -68,9 +68,6 @@
 #   USE_NNPACK=0
 #     disables NNPACK build
 #
-#   USE_QNNPACK=0
-#     disables QNNPACK build (quantized 8-bit operators)
-#
 #   USE_DISTRIBUTED=0
 #     disables distributed (c10d, gloo, mpi, etc.) build
 #
@@ -366,7 +363,6 @@ def get_submodule_folders():
             "cpuinfo",
             "onnx",
             "foxi",
-            "QNNPACK",
             "fbgemm",
             "cutlass",
         ]
