diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index c67746d903..b596f4728d 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -203,7 +203,7 @@ elseif(BLAS STREQUAL "MKL")
   if(BLAS_SET_BY_USER)
     find_package(MKL REQUIRED)
   else()
-    find_package(MKL QUIET)
+    find_package(MKL REQUIRED)
   endif()
   include(${CMAKE_CURRENT_LIST_DIR}/public/mkl.cmake)
   if(MKL_FOUND)
diff --git a/cmake/Modules/FindMKL.cmake b/cmake/Modules/FindMKL.cmake
index 01594a5b66..77230e2b92 100644
--- a/cmake/Modules/FindMKL.cmake
+++ b/cmake/Modules/FindMKL.cmake
@@ -170,7 +170,7 @@ ENDIF()
 
 MACRO(GET_MKL_LIB_NAMES LIBRARIES INTERFACE MKL64)
   cmake_parse_arguments("" "" "THREAD" "" ${ARGN})
-  SET(${LIBRARIES} mkl_${INTERFACE}${MKL64} mkl_core)
+  SET(${LIBRARIES} mkl_${INTERFACE}${MKL64} mkl_core_dll)
   IF(_THREAD)
     LIST(INSERT ${LIBRARIES} 1 ${_THREAD})
     IF(UNIX AND ${USE_STATIC_MKL})
@@ -324,7 +324,7 @@ IF (NOT "${MKL_THREADING}" STREQUAL "SEQ")
       FOREACH(mkl64 ${mkl64s} "")
         FOREACH(mklthread ${mklthreads})
           IF (NOT MKL_LIBRARIES)
-            GET_MKL_LIB_NAMES(mkl_lib_names "${mkliface}" "${mkl64}" THREAD "${mklthread}")
+            GET_MKL_LIB_NAMES(mkl_lib_names "${mkliface}" "${mkl64}_dll" THREAD "${mklthread}_dll")
             CHECK_ALL_LIBRARIES(MKL_LIBRARIES MKL_OPENMP_TYPE MKL_OPENMP_LIBRARY cblas_sgemm
               "${mkl_lib_names};${mklrtl};${mkl_pthread};${mkl_m};${mkl_dl}" "")
           ENDIF (NOT MKL_LIBRARIES)
