diff --git a/torch/utils/cpp_extension.py b/torch/utils/cpp_extension.py
index 11b233f27124e..aef5fd5793e3a 100644
--- a/torch/utils/cpp_extension.py
+++ b/torch/utils/cpp_extension.py
@@ -50,25 +50,10 @@
 # Or from include/crt/host_config.h in the CUDA SDK
 # The second value is the exclusive(!) upper bound, i.e. min <= version < max
 CUDA_GCC_VERSIONS: VersionMap = {
-    '11.0': (MINIMUM_GCC_VERSION, (10, 0)),
-    '11.1': (MINIMUM_GCC_VERSION, (11, 0)),
-    '11.2': (MINIMUM_GCC_VERSION, (11, 0)),
-    '11.3': (MINIMUM_GCC_VERSION, (11, 0)),
-    '11.4': ((6, 0, 0), (12, 0)),
-    '11.5': ((6, 0, 0), (12, 0)),
-    '11.6': ((6, 0, 0), (12, 0)),
-    '11.7': ((6, 0, 0), (12, 0)),
 }
 
 MINIMUM_CLANG_VERSION = (3, 3, 0)
 CUDA_CLANG_VERSIONS: VersionMap = {
-    '11.1': (MINIMUM_CLANG_VERSION, (11, 0)),
-    '11.2': (MINIMUM_CLANG_VERSION, (12, 0)),
-    '11.3': (MINIMUM_CLANG_VERSION, (12, 0)),
-    '11.4': (MINIMUM_CLANG_VERSION, (13, 0)),
-    '11.5': (MINIMUM_CLANG_VERSION, (13, 0)),
-    '11.6': (MINIMUM_CLANG_VERSION, (14, 0)),
-    '11.7': (MINIMUM_CLANG_VERSION, (14, 0)),
 }
 
 __all__ = ["get_default_build_root", "check_compiler_ok_for_platform", "get_compiler_abi_compatibility_and_version", "BuildExtension",

