diff --git a/torch/csrc/jit/python/script_init.cpp b/torch/csrc/jit/python/script_init.cpp
index 2f2f9f8785..c02d5ee135 100644
--- a/torch/csrc/jit/python/script_init.cpp
+++ b/torch/csrc/jit/python/script_init.cpp
@@ -1218,7 +1218,7 @@ void initJitScriptBindings(PyObject* module) {
              const std::string& method_name,
              const py::tuple& input_tuple) {
             Stack stack;
-            for (auto& input : input_tuple) {
+            for (const auto& input : input_tuple) {
               stack.push_back(toTypeInferredIValue(input));
             }
             return m.get_method(method_name)(stack);
@@ -1229,7 +1229,7 @@ void initJitScriptBindings(PyObject* module) {
           "forward",
           [](mobile::Module& m, const py::tuple& input_tuple) {
             Stack stack;
-            for (auto& input : input_tuple) {
+            for (const auto& input : input_tuple) {
               stack.push_back(toTypeInferredIValue(input));
             }
             return m.get_method("forward")(stack);
